### PR TITLE
perf: skip idle actors and batch effect expiry in updateWorldTime

### PR DESCRIPTION
**Pull request by ndguarino**
Every world-time change — which combat's next-turn/next-round buttons trigger on essentially every click — walked all game.actors plus current-scene tokens and serially awaited effect expiry, out-of-combat recovery, and continuing-charge checks per actor. In medium worlds each combat click was O(all actors) with multiple DB round-trips, tripping the hook's own >100 ms self-diagnostic.
 
Three structural changes:
- Pre-filter: actors with no temporary effects are skipped entirely unless a 12-second recovery boundary was crossed. Measured on a 44-actor world: steady-state ticks now process 9/44 actors in ~17 ms; the only >100 ms tick left is the segment-12 crossing, which is real work (recovery for all actors + actual adjustment fades).
- Batched writes: expireEffects now bulk-deletes expired effects (one deleteEmbeddedDocuments per parent) and posts one combined expiry chat message per actor, resolving the five standing TODO: bulk delete all AE's at once / one ChatMessage per actor comments.
- Serialized processing: Foundry doesn't await async hook callbacks, so rapid time advances could overlap and double-process the same effects (does not exist in the EmbeddedCollection errors). Ticks now queue on a promise chain.
 
Bug fixes found along the way: the secondsSinceRecovery remainder was zeroed on multi-recovery advances (a 25 s advance lost its banked 1 s); effects lacking system flags (user-created AEs) crashed expireEffects before their own deletion, making them immortal and crash-looping every tick; the hook's 2nd parameter was named options but is actually the v13 dt delta.

New quench batch HERO: World Time Recovery & Expiry (4 tests) covers the pre-filter edge cases, remainder banking, batched expiry, and the flags-less-effect fix. Full suite passes (the two powers.validate() failures are pre-existing on main from the STRIKE & TRIP xml entries).

### CODE REVIEW

File: module/utility/util.mjs [hunk 1]
```diff
@@ -237,6 +237,37 @@ export async function expireEffects(actor, expiresOn) {
...
+            // deleted some of these; deleting a missing id throws and would abort the batch.
+            const ids = Array.from(aeIds).filter((id) => parent.effects.has(id));
+            try {
+                if (ids.length > 0) await parent.deleteEmbeddedDocuments("ActiveEffect", ids);
+            } catch (e) {
+                console.error(`Failed to bulk delete expired effects on ${parent.uuid}: ${e.message}`);
+            }
+        }
+        if (expiredCardHtmls.length > 0) {
+            await ChatMessage.create({
```
Comment (ceguarino) [inline module/utility/util.mjs:261]: Needs to be a whisper

Comment (ceguarino) [reply module/utility/util.mjs:261]: Large number of GM only actors may be an issue here. Let's consider this.
